test/e2e_node/e2e_node_suite_test.go

```diff
@@ -124,45 +118,47 @@ var _ = SynchronizedBeforeSuite(func() []byte {
 	// We should mask locksmithd when provisioning the machine.
 	maskLocksmithdOnCoreos()
 
-	shared := &SharedContext{}
+	var data []byte
 	if *startServices {
-		e2es = newE2eService(framework.TestContext.NodeName, framework.TestContext.CgroupsPerQOS, framework.TestContext.EvictionHard, shared)
-		if err := e2es.start(); err != nil {
+		e2es = NewE2eServices()
+		manifestPath, err := e2es.StartE2EServices()
+		if err != nil {
 			Fail(fmt.Sprintf("Unable to start node services.\n%v", err))
 		}
+		data = []byte(manifestPath)
 		glog.Infof("Node services started.  Running tests...")
 	} else {
 		glog.Infof("Running tests without starting services.")
 	}
 
 	glog.Infof("Starting namespace controller")
+	// TODO(random-liu): Move namespace controller into namespace services.
 	startNamespaceController()
 
 	// Reference common test to make the import valid.
 	commontest.CurrentSuite = commontest.NodeE2E
 
-	// Share the node name with the other test nodes.
-	shared.NodeName = framework.TestContext.NodeName
-	data, err := json.Marshal(shared)
-	Expect(err).NotTo(HaveOccurred())
-
 	return data
 }, func(data []byte) {
-	// Set the shared context got from the synchronized initialize function
-	shared := &SharedContext{}
-	Expect(json.Unmarshal(data, shared)).To(Succeed())
-	context = *shared
+	// TODO(random-liu): Get manifest path from kubelet flags when we move kubelet start logic out of the test.
+	framework.TestContext.ManifestPath = string(data)
 
-	framework.TestContext.NodeName = shared.NodeName
+	// Initialize node name here to make sure every test node will get right node name.
+	if framework.TestContext.NodeName == "" {
```

Comment:
There are 2 functions in `SynchronizedBeforeSuite`. The first one is only called once, the second one is called on every parallel test node.

We need to initialize node name to start kubelet in the first function now.
However, only the variable initialized in the second function can be accessed by all test nodes, so we have to initialize it again in the second function.

After moving the kubelet start logic out, we can only keep the second initalization.